test/unit/math/test_ad_matvar.hpp

```diff
@@ -398,27 +388,50 @@ template <typename... Types, typename F, typename... EigMats>
 void expect_ad_matvar_impl(const ad_tolerances& tols, const F& f,
                            const EigMats&... x) {
   using stan::is_var;
+  using stan::is_var_matrix;
+  using stan::is_eigen;
   using stan::plain_type_t;
   using stan::math::promote_scalar_t;
   using stan::math::var;
   using stan::math::var_value;
   using stan::math::test::type_name;
+  
+  if(!stan::math::disjunction<is_var_matrix<Types>...>::value) {
+    FAIL() << "expect_ad_matvar requires at least one varmat input!" << std::endl;
+  }
+
+  if(!stan::math::disjunction<is_var<scalar_type_t<Types>>...>::value) {
+    FAIL() << "expect_ad_matvar requires at least one autodiff input!" << std::endl;
+  }
+
+  auto A_mv_tuple = std::make_tuple(convert_to_matvar<Types>(x)...);
+  auto A_vm_tuple = std::make_tuple(convert_to_varmat<Types>(x)...);
```

Comment:
Would a better name be `make_matvar/varmat_compatible`? Convert makes it sound like it's always going to transform to something else